### PR TITLE
Curve functions accept numeric rate of spread

### PR DESCRIPTION
**Pull request by macfarlandian**
## Description of the change

Both the R(t) feature and Excalibur are going to need to generate curves based on arbitrary rate of spread factors that aren't captured by our existing low/med/high enums. This PR changes the interface for our curve functions to just require numeric inputs. 

There was existing logic in the curve function to translate the enum key into numeric values; that has been factored out into a separate function. Components that were dumping the `EpidemicModelContext` directly into curve functions now need to use this new function to transform the context data structure first. Future components for the features mentioned above may provide these values by other means, which may or may not depend on the enum value selected by the user. There may come a point where we deprecate and remove that enum entirely, but we're not there yet, so for now this new function bridges the gap.

Also added some tests for the curve function to help me make this change with confidence. They mainly just replicate the manual sanity checks we have been doing on model inspection page (inspecting table structure, computing checksums).

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Part of #62 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/infection-model/seir.tsx [hunk 1]
```diff
@@ -272,15 +259,13 @@ export function getAllBracketCurves(inputs: CurveProjectionInputs) {
...
     rateOfSpreadCells -
     (1 - facilityOccupancyPct) *
       (rateOfSpreadCells - rateOfSpreadCellsAdjustment);
-  let rateOfSpreadDorms = R0Dorms[rateOfSpreadFactor];
   const rateOfSpreadDormsAdjustment = 1.7; // magic constant
-  rateOfSpreadDorms =
+  const rateOfSpreadDormsAdjusted =
     rateOfSpreadDorms -
     (1 - facilityOccupancyPct) *
       (rateOfSpreadDorms - rateOfSpreadDormsAdjustment);
```
Comment (macfarlandian) [inline src/infection-model/seir.tsx:271]: @justkunz it's not strictly in scope for this change but I was wondering whether you think all these adjustments will still be valid when the input values come from an R(t) calculation?

File: src/infection-model/index.tsx [hunk 2]
```diff
@@ -14,6 +15,14 @@ export type CurveData = {
   staff: ndarray;
 };
 
+export type CurveFunctionInputs = Omit<
```
Comment (jessex) [inline src/infection-model/index.tsx:18]: So this whole operation means that `CurveFunctionInputs` has all fields in `EpidemicModelInputs` except for `rateOfSpreadFactor` but also has `rateOfSpreadCells` and `rateOfSpreadDorms` added. Right?

**Comment by macfarlandian** [comment]
@jessex that's the gist of it, yes.

File: src/infection-model/index.tsx (hunk 2 above)
Comment (macfarlandian) [reply src/infection-model/index.tsx:18]: that is correct

**Pull request merged**